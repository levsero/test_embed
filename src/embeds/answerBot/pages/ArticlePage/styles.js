import styled from 'styled-components'
import { FONT_SIZE } from 'src/constants/shared'
import { SlideAppear } from 'component/transition/SlideAppear'

const FeedbackContainer = styled(SlideAppear)`
  width: 100%;
  position: absolute;
  overflow: hidden;
  border-radius: 0 0 ${8 / FONT_SIZE}rem ${8 / FONT_SIZE}rem;
`

export { FeedbackContainer }
