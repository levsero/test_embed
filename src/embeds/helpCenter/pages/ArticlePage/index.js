import React from 'react'
import { connect } from 'react-redux'
import PropTypes from 'prop-types'
import { Widget, Main, Header } from 'src/components/Widget'
import HelpCenterFooter from 'src/embeds/helpCenter/components/Footer'
import { i18n } from 'service/i18n'
import { getLocale } from 'src/redux/modules/base/base-selectors'
import {
  getActiveArticle,
  getRestrictedImages,
  getResultsLocale
} from 'src/embeds/helpCenter/selectors'
import { getSettingsHelpCenterOriginalArticleButton } from 'src/redux/modules/settings/settings-selectors'
import {
  handleOriginalArticleClicked,
  performImageSearch,
  addRestrictedImage
} from 'src/embeds/helpCenter/actions'
import {
  getSettingsHelpCenterTitle,
  getHelpCenterButtonLabel,
  getChatConnectionConnecting,
  getShowNextButton
} from 'src/redux/modules/selectors'
import HelpCenterArticle from 'src/components/HelpCenterArticle'
import { isMobileBrowser } from 'utility/devices'

const ArticlePage = ({
  activeArticle,
  addRestrictedImage,
  handleOriginalArticleClicked,
  isMobile,
  onClick,
  originalArticleButton,
  performImageSearch,
  restrictedImages,
  resultsLocale,
  title,
  showNextButton
}) => {
  return (
    <Widget>
      <Header title={title} />
      <Main>
        <HelpCenterArticle
          activeArticle={activeArticle}
          locale={resultsLocale}
          originalArticleButton={originalArticleButton}
          handleOriginalArticleClick={handleOriginalArticleClicked}
          storedImages={restrictedImages}
          imagesSender={performImageSearch}
          updateStoredImages={addRestrictedImage}
          isMobile={isMobile}
        />
      </Main>
      <HelpCenterFooter onClick={onClick} showNextButton={showNextButton} />
    </Widget>
  )
}

ArticlePage.propTypes = {
  activeArticle: PropTypes.object,
  originalArticleButton: PropTypes.bool,
  performImageSearch: PropTypes.func.isRequired,
  handleOriginalArticleClicked: PropTypes.func.isRequired,
  restrictedImages: PropTypes.objectOf(PropTypes.string).isRequired,
  addRestrictedImage: PropTypes.func,
  resultsLocale: PropTypes.string,
  title: PropTypes.string,
  isMobile: PropTypes.bool,
  onClick: PropTypes.func,
  showNextButton: PropTypes.bool
}

ArticlePage.defaultProps = {
  onClick: () => {}
}

const mapStateToProps = state => {
  const articleTitleKey = 'help'
  const titleKey = `embeddable_framework.helpCenter.form.title.${articleTitleKey}`

  return {
    locale: getLocale(state),
    activeArticle: getActiveArticle(state),
    originalArticleButton: getSettingsHelpCenterOriginalArticleButton(state),
    resultsLocale: getResultsLocale(state),
    restrictedImages: getRestrictedImages(state),
    title: getSettingsHelpCenterTitle(state, titleKey),
    buttonLabel: getHelpCenterButtonLabel(state),
    loading: getChatConnectionConnecting(state),
    isMobile: isMobileBrowser(),
    isRTL: i18n.isRTL(),
    showNextButton: getShowNextButton(state)
  }
}

const actionCreators = {
  handleOriginalArticleClicked,
  performImageSearch,
  addRestrictedImage
}

const connectedComponent = connect(
  mapStateToProps,
  actionCreators
)(ArticlePage)

export { connectedComponent as default, ArticlePage as Component }
