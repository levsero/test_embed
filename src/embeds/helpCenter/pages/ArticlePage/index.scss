@import 'src/styles/_vars.scss';

.button {
  padding-right: #{10 / $font-size}rem !important;
  padding-left: #{10 / $font-size}rem !important;
}
.buttonDesktop {
  max-width: #{200 / $font-size}rem !important;
}

.buttonMobile {
  width: 100%;
}

.buttonGroup {
  margin-top: #{15 / $font-size}rem;
}

.buttonGroupNoLogo {
  margin-bottom: #{15 / $font-size}rem;
}

.buttonContainer {
  position: relative !important;
}

.footerContentMultiple {
  display: flex;
  align-items: flex-end;
  justify-content: space-between;
}
