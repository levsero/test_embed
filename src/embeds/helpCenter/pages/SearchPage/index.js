import React, { useRef, useEffect } from 'react'
import { connect } from 'react-redux'
import PropTypes from 'prop-types'

import { Widget, Main } from 'src/components/Widget'
import SearchHeader from 'src/embeds/helpCenter/components/SearchHeader'
import HelpCenterFooter from 'src/embeds/helpCenter/components/Footer'
import Results from 'src/embeds/helpCenter/components/Results'

import {
  getHideZendeskLogo,
  getSettingsHelpCenterTitle,
  getShowNextButton
} from 'src/redux/modules/selectors'
import { getSearchLoading, getSearchedArticles } from 'embeds/helpCenter/selectors'
import { isMobileBrowser } from 'utility/devices'
import LoadingBarContent from 'src/components/LoadingBarContent'
import NotificationPopup from 'src/embeds/helpCenter/components/NotificationPopup'

const SearchPage = ({ title, showNextButton, isMobile, isSearchLoading, articles }) => {
  const searchHeaderRef = useRef(null)
  const content = isSearchLoading ? <LoadingBarContent /> : <Results />
  useEffect(() => {
    if (!articles.length) {
      searchHeaderRef.current.focus()
    }
  }, [articles])

  return (
    <Widget>
      <SearchHeader ref={searchHeaderRef} isMobile={isMobile} title={title} />
      <Main>{content}</Main>
      <HelpCenterFooter showNextButton={showNextButton} />
      {!isMobile && <NotificationPopup />}
    </Widget>
  )
}

SearchPage.propTypes = {
  showNextButton: PropTypes.bool,
  title: PropTypes.string.isRequired,
  isMobile: PropTypes.bool,
  isSearchLoading: PropTypes.bool,
  articles: PropTypes.array
}

const mapStateToProps = state => ({
  title: getSettingsHelpCenterTitle(state),
  showNextButton: getShowNextButton(state),
  isMobile: isMobileBrowser(),
  hideZendeskLogo: getHideZendeskLogo(state),
  isSearchLoading: getSearchLoading(state),
  articles: getSearchedArticles(state)
})

const connectedComponent = connect(mapStateToProps)(SearchPage)

export { connectedComponent as default, SearchPage as Component }
