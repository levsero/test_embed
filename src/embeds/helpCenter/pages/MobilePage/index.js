import React, { Component } from 'react'
import PropTypes from 'prop-types'

import { ScrollContainer } from 'component/container/ScrollContainer'
import { SearchField } from 'component/field/SearchField'
import { ZendeskLogo } from 'component/ZendeskLogo'
import { LoadingBarContent } from 'component/loading/LoadingBarContent'
import { i18n } from 'service/i18n'
import HelpCenterChannelButton from 'src/embeds/helpCenter/components/HelpCenterChannelButton'

import { locals as styles } from './styles.scss'

export default class MobilePage extends Component {
  static propTypes = {
    articleViewActive: PropTypes.bool,
    buttonLabel: PropTypes.string.isRequired,
    children: PropTypes.node.isRequired,
    hasContextualSearched: PropTypes.bool,
    handleNextClick: PropTypes.func.isRequired,
    handleOnChangeValue: PropTypes.func.isRequired,
    onSearchFieldFocus: PropTypes.func.isRequired,
    hasSearched: PropTypes.bool,
    hideZendeskLogo: PropTypes.bool,
    isLoading: PropTypes.bool,
    search: PropTypes.func.isRequired,
    searchFieldValue: PropTypes.string,
    showNextButton: PropTypes.bool,
    isContextualSearchPending: PropTypes.bool.isRequired,
    contextualHelpRequestNeeded: PropTypes.bool.isRequired,
    searchPlaceholder: PropTypes.string.isRequired,
    title: PropTypes.string.isRequired,
    buttonLoading: PropTypes.bool
  }

  static defaultProps = {
    articleViewActive: false,
    chatAvailable: false,
    hasContextualSearched: false,
    hasSearched: false,
    hideZendeskLogo: false,
    isLoading: false,
    searchFieldValue: '',
    showNextButton: true,
    submitTicketAvailable: true,
    chatEnabled: false,
    channelChoice: false,
    setChannelChoiceShown: () => {},
    talkOnline: false,
    buttonLoading: false
  }

  constructor(props, context) {
    super(props, context)

    this.state = {
      searchFieldFocused: false,
      showIntroScreen: !props.contextualHelpRequestNeeded && !props.articleViewActive
    }

    this.searchField = null
    this.searchFieldFocusTimer = null
  }

  componentDidUpdate = (prevProps, prevState) => {
    // We have to do this check in componentDidUpdate so that
    // the searchField is the most recent one and ios focuses
    // on the correct one.
    if (
      prevState.showIntroScreen === true &&
      this.state.showIntroScreen === false &&
      this.props.hasContextualSearched === false
    ) {
      this.searchField.focus()
    }

    if (this.searchField) {
      this.searchField.setState({ searchInputVal: this.props.searchFieldValue })
    }
  }

  componentWillUnmount = () => {
    if (this.searchFieldFocusTimer) {
      clearTimeout(this.searchFieldFocusTimer)
    }
  }

  focusField() {
    this.getSearchField().focus()
  }

  getSearchField() {
    return this.searchField
  }

  resetState = () => {
    if (!this.props.hasSearched) {
      this.setState({ showIntroScreen: true })
      this.setSearchFieldFocused(false)
    }
  }

  setIntroScreen = () => {
    this.setState({
      showIntroScreen: false
    })
  }

  setSearchFieldFocused = focused => {
    this.setState({ searchFieldFocused: !!focused })
    this.props.onSearchFieldFocus(!!focused)
  }

  handleSearchBoxClicked = () => {
    if (this.state.showIntroScreen) {
      this.setState({ showIntroScreen: false })
      this.setSearchFieldFocused(true)
    }
  }

  handleOnBlur = () => {
    // defer event to allow onClick events to fire first
    this.searchFieldFocusTimer = setTimeout(() => {
      this.setSearchFieldFocused(false)

      if (!this.props.hasSearched && !this.props.isLoading) {
        this.setState({ showIntroScreen: true })
      }
    }, 1)
  }

  handleOnFocus = () => {
    this.setSearchFieldFocused(true)
  }

  handleSubmit = e => {
    e.preventDefault()
    this.props.search()
  }

  showFooterContent = () => {
    return (
      this.props.showNextButton &&
      (this.props.articleViewActive ||
        (!this.state.showIntroScreen && !this.state.searchFieldFocused))
    )
  }

  renderSearchField = () => {
    return (
      <SearchField
        ref={el => {
          this.searchField = el
        }}
        fullscreen={true}
        isMobile={true}
        onFocus={this.handleOnFocus}
        onBlur={this.handleOnBlur}
        onChangeValue={this.props.handleOnChangeValue}
        hasSearched={this.props.hasSearched}
        onSearchIconClick={this.handleSubmit}
        onClick={this.handleSearchBoxClicked}
        isLoading={this.props.isLoading}
        searchPlaceholder={this.props.searchPlaceholder}
      />
    )
  }

  renderForm = () => {
    const hiddenClasses = !this.state.showIntroScreen ? 'u-isHidden' : ''

    return (
      <form
        ref="helpCenterForm"
        className={styles.form}
        noValidate={true}
        onSubmit={this.handleSubmit}
      >
        <h1 className={`${styles.searchTitle} ${hiddenClasses}`}>
          {i18n.t('embeddable_framework.helpCenter.label.searchHelpCenter')}
        </h1>
        {this.renderSearchField()}
      </form>
    )
  }

  renderFormContainer = () => {
    return this.props.articleViewActive || !this.state.showIntroScreen ? null : (
      <div>{this.renderForm()}</div>
    )
  }

  renderHeaderContent = () => {
    return this.props.articleViewActive || this.state.showIntroScreen ? null : this.renderForm()
  }

  renderFooterContent = () => {
    if (!this.showFooterContent()) return null

    const { buttonLabel, handleNextClick, buttonLoading } = this.props
    return (
      <HelpCenterChannelButton
        buttonLabel={buttonLabel}
        onClick={handleNextClick}
        loading={buttonLoading}
        isRTL={i18n.isRTL()}
        isMobile={true}
      />
    )
  }

  renderZendeskLogo = hideZendeskLogo => {
    return !hideZendeskLogo ? <ZendeskLogo fullscreen={true} /> : null
  }

  renderChildContent() {
    const { children, isContextualSearchPending, articleViewActive } = this.props

    if (this.state.showIntroScreen && !articleViewActive) return null

    return isContextualSearchPending && !articleViewActive ? (
      <LoadingBarContent containerClasses={styles.loadingBarContent} />
    ) : (
      children
    )
  }

  render = () => {
    const mobileHideLogoState = this.props.hasSearched
    const hideZendeskLogo = this.props.hideZendeskLogo || mobileHideLogoState
    const containerClasses = !this.props.showNextButton && hideZendeskLogo ? styles.container : ''

    return (
      <div>
        <ScrollContainer
          ref="scrollContainer"
          title={this.props.title}
          headerContent={this.renderHeaderContent()}
          footerContent={this.renderFooterContent()}
          fullscreen={true}
          isMobile={true}
          containerClasses={containerClasses}
          isVirtualKeyboardOpen={this.state.searchFieldFocused}
        >
          {this.renderFormContainer()}
          {this.renderChildContent()}
        </ScrollContainer>
        {this.renderZendeskLogo(hideZendeskLogo)}
      </div>
    )
  }
}
