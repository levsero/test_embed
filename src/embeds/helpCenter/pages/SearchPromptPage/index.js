import React, { useEffect, useRef } from 'react'
import { connect } from 'react-redux'
import PropTypes from 'prop-types'

import { Widget, Main, Footer } from 'src/components/Widget'
import WidgetHeader from 'src/components/WidgetHeader'
import ZendeskLogo from 'src/components/ZendeskLogo'
import SearchForm from 'src/embeds/helpCenter/components/SearchForm'
import { locals as styles } from './styles.scss'
import { getHideZendeskLogo, getSettingsHelpCenterTitle } from 'src/redux/modules/selectors'
import { isMobileBrowser } from 'utility/devices'
import { performSearch } from 'embeds/helpCenter/actions'
import { i18n } from 'service/i18n'

const SearchPromptPage = ({ title, hideZendeskLogo, isMobile, header }) => {
  const searchFormRef = useRef(null)
  useEffect(() => {
    searchFormRef.current.focus()
  }, [])
  return (
    <Widget>
      <WidgetHeader>{title}</WidgetHeader>
      <Main>
        {isMobile && <h1 className={styles.title}>{header}</h1>}
        <SearchForm ref={searchFormRef} />
      </Main>
      <Footer>
        {!hideZendeskLogo && (
          <div className={styles.footer}>
            <ZendeskLogo />
          </div>
        )}
      </Footer>
    </Widget>
  )
}

SearchPromptPage.propTypes = {
  header: PropTypes.string,
  isMobile: PropTypes.bool,
  title: PropTypes.string.isRequired,
  hideZendeskLogo: PropTypes.bool.isRequired
}

const mapStateToProps = state => ({
  isMobile: isMobileBrowser(),
  title: getSettingsHelpCenterTitle(state),
  header: i18n.t('embeddable_framework.helpCenter.label.searchHelpCenter'),
  hideZendeskLogo: getHideZendeskLogo(state)
})

const mapDispatchToProps = {
  performSearch
}

const connectedComponent = connect(
  mapStateToProps,
  mapDispatchToProps
)(SearchPromptPage)

export { connectedComponent as default, SearchPromptPage as Component }
