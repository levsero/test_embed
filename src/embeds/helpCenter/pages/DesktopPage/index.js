import React, { Component } from 'react'
import PropTypes from 'prop-types'

import { Button } from '@zendeskgarden/react-buttons'
import { ButtonGroup } from 'component/button/ButtonGroup'
import { ScrollContainer } from 'component/container/ScrollContainer'
import { SearchField } from 'component/field/SearchField'
import { ZendeskLogo } from 'component/ZendeskLogo'
import { LoadingBarContent } from 'component/loading/LoadingBarContent'
import { i18n } from 'service/i18n'
import { LoadingEllipses } from 'component/loading/LoadingEllipses'

import SearchPromptPage from 'src/embeds/helpCenter/pages/SearchPromptPage'
import { locals as styles } from './styles.scss'
import classNames from 'classnames'

export default class DesktopPage extends Component {
  static propTypes = {
    articleViewActive: PropTypes.bool,
    buttonLabel: PropTypes.string.isRequired,
    channelChoice: PropTypes.bool,
    children: PropTypes.node.isRequired,
    handleNextClick: PropTypes.func.isRequired,
    handleOnChangeValue: PropTypes.func.isRequired,
    hasSearched: PropTypes.bool,
    hideZendeskLogo: PropTypes.bool,
    isLoading: PropTypes.bool,
    onNextClick: PropTypes.func,
    search: PropTypes.func.isRequired,
    searchFieldValue: PropTypes.string,
    showNextButton: PropTypes.bool,
    isContextualSearchPending: PropTypes.bool.isRequired,
    isOnInitialDesktopSearchScreen: PropTypes.bool,
    maxWidgetHeight: PropTypes.oneOfType([PropTypes.number, PropTypes.object]),
    searchPlaceholder: PropTypes.string.isRequired,
    title: PropTypes.string.isRequired,
    buttonLoading: PropTypes.bool,
    contextualHelpRequestNeeded: PropTypes.bool
  }

  static defaultProps = {
    articleViewActive: false,
    channelChoice: false,
    channelChoiceShown: false,
    formTitleKey: 'help',
    hasSearched: false,
    hideZendeskLogo: false,
    isLoading: false,
    onNextClick: () => {},
    searchFieldValue: '',
    showNextButton: true,
    submitTicketAvailable: true,
    chatEnabled: false,
    isOnInitialDesktopSearchScreen: false,
    buttonLoading: false,
    contextualHelpRequestNeeded: false
  }

  constructor(props, context) {
    super(props, context)

    this.agentMessage = null
    this.searchField = null
  }

  componentDidUpdate = () => {
    if (this.searchField) {
      this.searchField.setState({
        searchInputVal: this.props.searchFieldValue
      })
    }
  }

  getSearchField() {
    return this.searchField
  }

  focusField = () => {
    if (!this.props.articleViewActive && this.searchField) {
      const searchFieldInputNode = this.searchField.getSearchField()
      const strLength = searchFieldInputNode.value.length

      this.searchField.focus()

      if (searchFieldInputNode.setSelectionRange) {
        searchFieldInputNode.setSelectionRange(strLength, strLength)
      }
    }
  }

  handleSubmit = e => {
    e.preventDefault()
    this.props.search()
  }

  renderForm = () => {
    const { hasSearched, contextualHelpRequestNeeded, hideZendeskLogo } = this.props

    const customSearchContainerClasses = classNames({
      [styles.onHelpCenterSmallScreen]:
        !hasSearched && !contextualHelpRequestNeeded && hideZendeskLogo,
      [styles.onHelpCenterLargeScreen]: hasSearched || contextualHelpRequestNeeded
    })

    return (
      <form
        ref="helpCenterForm"
        noValidate={true}
        className={styles.form}
        onSubmit={this.handleSubmit}
      >
        <SearchField
          ref={el => {
            this.searchField = el
          }}
          fullscreen={false}
          hideZendeskLogo={hideZendeskLogo}
          onChangeValue={this.props.handleOnChangeValue}
          hasSearched={this.props.hasSearched}
          onSearchIconClick={this.handleSubmit}
          isLoading={this.props.isLoading}
          searchPlaceholder={this.props.searchPlaceholder}
          customSearchContainerClasses={customSearchContainerClasses}
        />
      </form>
    )
  }

  renderHeaderContent = () => {
    return this.props.isOnInitialDesktopSearchScreen || this.props.articleViewActive
      ? null
      : this.renderForm()
  }

  renderBodyForm = () => {
    return !this.props.isOnInitialDesktopSearchScreen ? null : this.renderForm()
  }

  renderZendeskLogo = () => {
    return !this.props.hideZendeskLogo ? <ZendeskLogo fullscreen={false} /> : null
  }

  renderLoadingAnimation = () => {
    return <LoadingEllipses useUserColor={false} itemClassName={styles.loadingAnimation} />
  }

  renderLoadingButton = () => {
    const buttonStyles = classNames(styles.button, styles.disabledButton)

    return (
      <Button primary={true} className={buttonStyles}>
        {this.renderLoadingAnimation()}
      </Button>
    )
  }

  renderButton = () => {
    const { channelChoice, buttonLabel, onNextClick, handleNextClick } = this.props
    const onClickHandler = channelChoice ? onNextClick : handleNextClick

    return (
      <Button primary={true} onClick={onClickHandler} className={styles.button}>
        {buttonLabel}
      </Button>
    )
  }

  renderFooterContent = () => {
    const { showNextButton, buttonLoading } = this.props

    return showNextButton && !this.props.isOnInitialDesktopSearchScreen ? (
      <div className={styles.buttonContainer}>
        <ButtonGroup rtl={i18n.isRTL()} containerClasses={styles.buttonGroup}>
          {buttonLoading ? this.renderLoadingButton() : this.renderButton()}
        </ButtonGroup>
      </div>
    ) : null
  }

  renderChildContent() {
    const { children, isContextualSearchPending, articleViewActive } = this.props

    return isContextualSearchPending && !articleViewActive ? (
      <LoadingBarContent containerClasses={styles.loadingBars} />
    ) : (
      children
    )
  }

  render = () => {
    const customHeightClasses = this.props.isOnInitialDesktopSearchScreen
      ? styles.noCustomHeight
      : ''
    let footerClasses = ''

    if (!this.props.showNextButton) {
      if (this.props.articleViewActive && this.props.hideZendeskLogo) {
        footerClasses = styles.footerArticleView
      } else {
        footerClasses = this.props.hideZendeskLogo ? styles.footer : styles.footerLogo
      }
    }
    const shadowVisible =
      this.props.showNextButton &&
      !this.props.isOnInitialDesktopSearchScreen &&
      !this.props.hideZendeskLogo

    if (this.props.isOnInitialDesktopSearchScreen) {
      return (
        <SearchPromptPage
          title={this.props.title}
          makeSearchRequest={this.props.search}
          handleOnChangeValue={this.props.handleOnChangeValue}
          isLoading={this.props.isLoading}
          searchPlaceholder={this.props.searchPlaceholder}
        />
      )
    }

    return (
      <div>
        <ScrollContainer
          ref="scrollContainer"
          hideZendeskLogo={this.props.hideZendeskLogo}
          title={this.props.title}
          classes={customHeightClasses}
          maxHeight={this.props.maxWidgetHeight}
          footerClasses={footerClasses}
          scrollShadowVisible={shadowVisible}
          headerContent={this.renderHeaderContent()}
          footerContent={this.renderFooterContent()}
        >
          {this.renderBodyForm()}
          {this.renderChildContent()}
        </ScrollContainer>
        {this.renderZendeskLogo()}
      </div>
    )
  }
}
