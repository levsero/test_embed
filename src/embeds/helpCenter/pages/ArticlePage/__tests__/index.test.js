import { Component as ArticlePage } from 'src/embeds/helpCenter/pages/ArticlePage'
import React from 'react'
import { render, fireEvent } from '@testing-library/react'
import { Provider } from 'react-redux'
import createStore from 'src/redux/createStore'

const performImageSearch = jest.fn(),
  handleOriginalArticleClicked = jest.fn(),
  onClick = jest.fn()

const renderComponent = inProps => {
  const props = {
    buttonLabel: 'buttonLabel',
    activeArticle: { title: 'articleTitle' },
    originalArticleButton: false,
    performImageSearch,
    handleOriginalArticleClicked,
    restrictedImages: {},
    resultsLocale: 'derp',
    title: 'pageTitle',
    isMobile: false,
    hideZendeskLogo: false,
    loading: false,
    showNextButton: true,
    onClick,
    ...inProps
  }

  return render(
    <Provider store={createStore()}>
      <ArticlePage {...props} />
    </Provider>
  )
}

describe('default render', () => {
  it('matches snapshot', () => {
    expect(renderComponent().container).toMatchSnapshot()
  })

  it('renders button label', () => {
    expect(renderComponent().getByText('buttonLabel')).toBeInTheDocument()
  })

  it('renders the article', () => {
    expect(renderComponent().getByText('articleTitle')).toBeInTheDocument()
  })

  it('renders page title', () => {
    expect(renderComponent().getByText('pageTitle')).toBeInTheDocument()
  })

  it('renders zendesk logo', () => {
    expect(renderComponent().container).toMatchSnapshot()
  })
})

describe('when hiding zendeskLogo', () => {
  it('does not render the zendesk logo', () => {
    const result = renderComponent({ hideZendeskLogo: true })
    expect(result.queryByText('ZendeskLogo')).toBeNull()
  })
})

describe('button', () => {
  it('onClick', () => {
    const { getByText } = renderComponent()

    fireEvent.click(getByText('buttonLabel'))
    expect(onClick).toHaveBeenCalled()

    onClick.mockReset()
  })

  describe('when disabled', () => {
    it('does not render', () => {
      const { queryByText } = renderComponent({ showNextButton: false })

      expect(queryByText('buttonLabel')).toBeNull()
    })
  })
})
