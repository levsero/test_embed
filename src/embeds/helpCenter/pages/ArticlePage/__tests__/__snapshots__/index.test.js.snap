// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`default render matches snapshot 1`] = `
.c0 {
  border-radius: 0.5714285714285714rem 0.5714285714285714rem 0 0;
  padding: 0.5714285714285714rem 1rem;
  background: !important;
  color: !important;
}

.c3 {
  font-weight: 700;
  text-align: center;
  -webkit-letter-spacing: 0.02142857142857143rem;
  -moz-letter-spacing: 0.02142857142857143rem;
  -ms-letter-spacing: 0.02142857142857143rem;
  letter-spacing: 0.02142857142857143rem;
  line-height: 1.5;
  font-size: 1.0714285714285714rem;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  margin-bottom: 2px;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  min-height: 2rem;
}

.c1:not(:first-child) {
  margin-top: 0.2857142857142857rem;
}

.c1 .c2:first-child {
  margin-left: 4rem;
}

.c1 .c2:nth-child(2) {
  margin-left: 2rem;
}

.c1 .c2:last-child {
  margin-right: 4rem;
}

.c1 .c2:nth-last-child(2) {
  margin-right: 2rem;
}

.c1 .c4 {
  width: 2rem !important;
  height: 2rem !important;
}

.c5 {
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c5 svg {
  width: 1rem !important;
  height: 1rem !important;
}

.c5 svg,
.c5 path {
  fill: #fff !important;
  color: #fff !important;
}

.c5:focus {
  outline: none !important;
  box-shadow: 0 0 0 0.21428571428571427rem rgba(255,255,255,0.4) !important;
}

.c7 {
  position: relative !important;
  margin-top: 1.0714285714285714rem;
}

.c8 {
  padding-right: 0.7142857142857143rem !important;
  padding-left: 0.7142857142857143rem !important;
  max-width: 14.285714285714286rem;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

<div>
  <div
    class="container"
  >
    <div
      class="c0"
    >
      <div
        class="c1"
      >
        <h1
          class="c2 c3"
        >
          pageTitle
        </h1>
        <button
          aria-label="Minimize"
          class="c4 index__c-btn--icon___1Yl4x c5 index__c-btn___3y2Qe index__c-btn--basic___38OO7 index__c-btn--muted___ze6U0 index__c-btn--sm___3Kuhv "
          data-garden-id="buttons.icon_button"
          data-garden-version="6.4.4"
          tabindex="0"
          type="button"
        >
          <svg
            class="index__c-btn__icon___3V0Rl"
            data-garden-id="buttons.icon"
            data-garden-version="6.4.4"
            data-testid="Icon--dash"
            realfilename="dash-fill.svg"
          />
        </button>
      </div>
    </div>
    <div
      class="container"
      data-testid="scroll-container-content"
    >
      <div
        class="content "
        data-testid="hc-article"
        lang="derp"
      >
        <h2
          class="title"
          data-testid="hc-article-title"
        >
          articleTitle
        </h2>
        <div
          class="article"
          data-testid="hc-article-body"
          role="presentation"
        />
      </div>
    </div>
    <footer
      class="footer footerShadow"
    >
      <div
        class="c6"
      >
        <a
          class="icon"
          data-testid="Icon--zendesk"
          href="https://www.zendesk.com/embeddables/?utm_source=webwidget&utm_medium=poweredbyzendesk&utm_campaign=image"
          tabindex="-1"
          target="_blank"
        >
          <svg
            realfilename="widget-icon_zendesk.svg"
            title="zendesk"
          />
        </a>
        <div
          class="c7"
        >
          <div
            class="container buttonRight "
          >
            <button
              class="c8 index__c-btn___3y2Qe index__c-btn--primary___1YOfW "
              data-garden-id="buttons.button"
              data-garden-version="6.4.4"
              tabindex="0"
              type="button"
            >
              Leave us a message
            </button>
          </div>
        </div>
      </div>
    </footer>
  </div>
</div>
`;
